src/compiler/builderState.ts

```diff
@@ -361,13 +426,18 @@ namespace ts {
          */
         function updateExportedModules(sourceFile: SourceFile, exportedModulesFromDeclarationEmit: ExportedModulesFromDeclarationEmit | undefined, exportedModulesMapCache: ComputingExportedModulesMap) {
             if (!exportedModulesFromDeclarationEmit) {
-                exportedModulesMapCache.set(sourceFile.resolvedPath, false);
+                exportedModulesMapCache.nonExporting.add(sourceFile.resolvedPath);
                 return;
             }
 
             let exportedModules: Set<Path> | undefined;
             exportedModulesFromDeclarationEmit.forEach(symbol => addExportedModule(getReferencedFileFromImportedModuleSymbol(symbol)));
-            exportedModulesMapCache.set(sourceFile.resolvedPath, exportedModules || false);
+            if (exportedModules) {
+                exportedModulesMapCache.exporting.set(sourceFile.resolvedPath, exportedModules);

```

Comment:
eg. here we are not ensuring that nonExporting value if present is not deleted?

Reply:
We now track `nonExporting` as the deleted set of the map, which is kept in sync with the map contents.